library/extractor/src/main/java/com/google/android/exoplayer2/metadata/id3/TextInformationFrame.java

```diff
@@ -19,52 +19,77 @@
 
 import android.os.Parcel;
 import android.os.Parcelable;
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import com.google.android.exoplayer2.MediaMetadata;
 import com.google.android.exoplayer2.util.Util;
+import com.google.common.collect.ImmutableList;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 /** Text information ID3 frame. */
 public final class TextInformationFrame extends Id3Frame {
-
   @Nullable public final String description;
+
+  /** @deprecated Use {@code values[0]} instead. */
+  @Deprecated
   public final String value;
 
-  public TextInformationFrame(String id, @Nullable String description, String value) {
+  @NonNull
+  public final List<String> values;
+
+  public TextInformationFrame(String id, @Nullable String description, @NonNull List<String> values) {
     super(id);
     this.description = description;
-    this.value = value;
+    
+    if( values.size() > 0 ) {
```

Comment:
Given that the spec says (emphasis mine):
> All ID3v2 frames consists of one frame header followed by **one or more** fields containing the actual information.

I think we should make this type match the spec, and reject an empty list with `IllegalArgumentException`

Then where we construct the list in `Id3Decoder` we can do the transform from 'no field' -> `[""]` before passing it in here, and it's part of the existing handling 'invalid' files logic (similar to how the current code uses `decodeStringIfValid` inside `Id3Decoder` instead of trying to do that transform in this constructor).

Reply:
Resolved in 85c4d98.